python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -580,34 +574,60 @@ def do_evaluate(
             child.evaluate(df, context=context, mapping=mapping)
             for child in self.children
         ]
+        # Kleene logic for Any (OR) and All (AND) if ignore_nulls is
```

Comment:
Do you think it would be reasonable to combine the Any and All branches? The truth tables obviously differ on the off-diagonal elements involving nulls, but the inversion is simply handled by whether h_result is negated or not. AFAIK the two branches are therefore identical other than that, and the choice of plc aggregation. The code here is complex enough that deduplicating seems worthwhile.

Reply:
I'll have a go

Reply:
If it winds up making things worse feel free to revert.

Reply:
Done.